### DISCUSSION TITLE
"feature-proposal" vs "implementation-proposal"

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#2937] [General]
or "Want-to-be-implemented" vs "Want-to-implement"

When people propose a feature / make a feature request, it can one of 2 cases:
1. They want it but don't plan to implement it themselves. Either b/c of no skills or no time. They just suggest it to (other) developers.
2. They want it and are ready/plan/would-like to implement it. 

It may be useful to require people to clearly state which case it is, what do you think? 
Ideally to select a subcategory of issue to make; or place a label themselves.  Alternatively, maintainters to label it clearly as such. 

I think both developers, and non-developers, may  benefit from being able to search/browse by this criterion.
For instance , so that developers can find out whether somebody else is already working or planinig to work on what they want, before starting to work on a fork and pull request. Or can join/help the ones who are planning to.

EDIT: perhaps just require that category 1 be placed in the **Discussions** section, "Ideas" category, as opposed to in **Issues**

**Comment by xiota** [comment]
1. Who would police the "requirement" for people to categorize themselves according to intent?
2. What if plans or schedules change?  There are lots of examples in comments of people who said they would work on something but didn't.
3. Different people can implement features differently.  One implementation may be a better fit for the project than another.

**Comment by elextr** [comment]
Agree with what @xiota said.

Also there have been discussions about using a github template for issues, but there was never any real agreement on how simple/complex that should be.  I don't know of any public projects where templates work well, they are a feature more for private repositories that have a limited circle of users.
